bundles/org.openhab.binding.tado/src/main/resources/OH-INF/thing/thing-types.xml

```diff
@@ -131,23 +132,23 @@
 		<label>Temperature</label>
 		<description>Current temperature</description>
 		<category>Temperature</category>
-		<state readOnly="true" pattern="%.1f %unit%"></state>
+		<state readOnly="true" pattern="%.1f %unit%"/>
 	</channel-type>
 
 	<channel-type id="humidity">
-		<item-type>Number</item-type>
+		<item-type>Number:Dimensionless</item-type>
 		<label>Humidity</label>
 		<description>Current humidity in %</description>
 		<category>Humidity</category>
-		<state readOnly="true" pattern="%.1f %%"></state>
+		<state readOnly="true" pattern="%.1f %%"/>
 	</channel-type>
 
 	<channel-type id="heatingPower">
-		<item-type>Number</item-type>
+		<item-type>Number:Dimensionless</item-type>
 		<label>Heating Power</label>
 		<description>Current heating power</description>
 		<category>Fire</category>
-		<state readOnly="true" pattern="%.0f %%"></state>
+		<state readOnly="true" pattern="%.0f %% r.H."/>
```

Comment:
I'm not sure you can have anything after the unit, does it actually work?

Reply:
> not sure you can have anything after the unit, does it actually work? 

No. It doesn't work. :(

So -- contrary the opinion of @rkoshak -- this strengthens the argument for https://github.com/openhab/openhab-core/issues/4155

Reply:
It seems one can set it on an item manually. But not when it is set as a pattern in the thing xml. Or?

Reply:
PS it may be related to the UI change made by @mherwege where the UI seems to ignore the pattern from the thing xml, and just takes the default unit instead. Or??

Reply:
There should be no difference. If it can't be set from the thing.xml it needs to be fixed so that it can be set from there too. It should be consistent. 

I can't speak to the UI change.

Reply:
> can't speak to the UI change.

I am hoping that @mherwege can.

Reply:
The unit always had to be the last part of the state description.
There is a `UnitUtils` class in core that has a method to parse the unit from a provided pattern string (any source of a state description). The javadoc for it is clear, see: https://github.com/openhab/openhab-core/blob/c9ee9920478c783ea4f40dacab79ea41569ef505/bundles/org.openhab.core/src/main/java/org/openhab/core/types/util/UnitUtils.java#L157
This has been the case for more then 5 years.

I am not sure if you put something else the unit conversions still work. It may just show the Number as is in the default unit and append the last part of the state description. And that could just be the UI doing it.

Reply:
> > can't speak to the UI change.
> 
> I am hoping that @mherwege can.

I doubt it. As pointed out, the unit has to be the last part without blanks.

Reply:
> Works for me in 4.2 M1 if I set the pattern myself.
> 
> ```
> %.0f %% r.H.
> ```
> 
> <img alt="image" width="727" src="https://private-user-images.githubusercontent.com/6385096/313790338-54003a65-aceb-4466-a16a-87b89dff8e70.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA3OTc1NjIsIm5iZiI6MTcxMDc5NzI2MiwicGF0aCI6Ii82Mzg1MDk2LzMxMzc5MDMzOC01NDAwM2E2NS1hY2ViLTQ0NjYtYTE2YS04N2I4OWRmZjhlNzAucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMxOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMThUMjEyNzQyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MTk3NDhjOTg0NDg4NjA4NGQ3Y2ZiMDNhYmYwMGY5NDBmZWI4MDZlMDk4OGE1YjkxNzIyZmRkMzU5ZDBlODQyNyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.CUrCFEZbakknhPDiRSW1-63UuPxyV2MGmPk-4JH0Lwc">
> If it doesn't work in the snapshots or if it doesn't work when the pattern is set from an add-on that's a regression.

That is just setting a pattern, not a unit. The used value would be in the unit set for the item. No conversion would be applied.

Reply:
> That is just setting a pattern, not a unit. The used value would be in the unit set for the item. No conversion would be applied.

But that's all that @andrewfg is trying to do here too. They are pushing a state description pattern with stuff after the unit for display purposes. This whole discussion is about setting the pattern, not the unit itself.

Reply:
If your item would have a unit of %, the value would likely be correct. If the unit is ONE, the value would not be converted to % using the pattern `%.0f %% r.H`. It would with a pattern of `%.0f %%`. The pattern contains the unit for visualisation and drives the conversion for visualisation.

Reply:
> The pattern contains the unit for visualisation and drives the conversion for visualisation.

@mherwege if I understand your statement correctly, then I think your statement is wrong. What is the point of explicitly specifying the state pattern in the thing.xml file if the UI will always over write that pattern?

To illustrate the issue please see below actual example where the state pattern is explicitly "%.1f %% r.H." yet the UI offers "%.0f %unit%". I think that _if the binding designer has explicitly specified a state pattern_ in then the UI **_must_** use that pattern. (By contrast it is OK for the UI to offer an own suggestion when the thing.xml does not explicitly specify a state pattern).

**Thing channel definition in the thing.xml..**
```
	<channel-type id="humidity">
		<item-type>Number:Dimensionless</item-type>
		<label>Humidity</label>
		<description>Current humidity in %</description>
		<category>Humidity</category>
		<state readOnly="true" pattern="%.1f %% r.H."/>
	</channel-type>
```
**Binding Java code**
```
    public State getHumidity() {
        PercentageDataPoint humidity = zoneState.getSensorDataPoints().getHumidity();
        return humidity != null ? new QuantityType<>(humidity.getPercentage(), Units.PERCENT) : UnDefType.UNDEF;
    }
```
**Main UI screen**
![image](https://github.com/openhab/openhab-addons/assets/893994/48829190-dc4d-4029-b6dc-b07bef04893e)

Reply:
Your fundamental problem is that a state pattern on a Number item with dimension cannot contain anything after the unit. The unit has to be a string without blanks. That has always been the case. If you still want to add something after it, you should link a pure Number item and the whole configuration of the channel as a Number with dimension becomes meaningless.
Before my unit UI change, the state description pattern in the UI was always showing `%.0f unit` where unit was dynamically determined in the UI to be the default unit. In the UI unit PR, I did replace that with `%.0f %unit%`, which is more correct as it uses the configured unit, instead of the default one, and doesn't require changing when the unit would change. The UI has never been showing the state pattern from the channel. Note that this is only shown and relevant (and set) when using a Number item with dimension. If you do not change the default State Description Pattern in the UI, it will also not get saved in metadata and the default is kept.
I have seen your suggestion to make this a separate dimension and unit. A fundamental problem there would also be that you cannot have a unit consisting of 2 strings with a blank in between.

Reply:
> you cannot have a unit consisting of 2 strings with a blank in between

Just for the avoidance of doubt, are you referring to "10Kg" vs "10 Kg"? Or something else?

> This is not supported.

Ok. Perhaps we can/should add the word 'yet' to your statement? => WDYT? I would be happy to explore making a PR to that effect..

Reply:
I do see the UI does not show the state pattern from the channel type as a default when creating a new item on a channel. It has never done that. That could be a relevant enhancement.

Reply:
> Just for the avoidance of doubt, are you referring to "10Kg" vs "10 Kg"? Or something else?

No, it has to be `10 Kg` with the blank, but in `10 % r.H.`, the `% r.H.` could never be a unit, as it has a blank in it. And the code, when trying to find a unit, would try to interpret `r.H.` as the unit (and fail, so ignore units and use the pattern as is without unit conversion).

> WDYT? I would be happy to explore making a PR to that effect..

I think it is a very challenging thing to try. I have tried when I developed the input field functionality in BasicUI, Android app and mainUI. I couldn't get it to work reasonably. The challenge is that it become impossible to know what the unit is in a longer string if you don't use a convention. And the convention at the moment is that it is the end of the string. If you try, you have to consider and test it in all UI's.

Reply:
> I do see the UI does not show the state pattern from the channel type as a default when creating a new item on a channel. It has never done that. That could be a relevant enhancement.

So, again for the avoidance of doubt, are you saying that (putting aside this discussion about the specific case of the humidity channel on this binding), the state pattern from the thing.xml has **always been ignored** by the UI? If so, then yes, it should definitely be done; and not marked as an 'enhancement' but as a 'bug fix'.

Reply:
> And the code, when trying to find a unit, would try to interpret r.H. as the unit (and fail, so ignore units and use the pattern as is without unit conversion).

It should be technically feasible though to treat the first "word" after the "%.0f" (or what ever) as the unit and just pass through everything after that as is though, right? You've already mentioned that you can't have a space in a unit so if the convention is "%<format> <unit>[ <passthrough>]" that seems both feasible and not a breaking change. The `<format>` is the String formatter definition, `<unit>` is the unit, still no spaces allowed, and if there is a space after the `unit` everything after the unit is just passed through.

Reply:
For representation, that works. It breaks input. If you have a format string with passthrough, the passthrough should be ignored for input. But the pattern is not evaluated (or even available) at the time of interpreting the input string. There is already a lot of logic in the input treatment to try to even cope with date formats, numeric dot and/or comma in a reasonable way, let alone make this work as well.
Everything is of course possible in the end. But I think it is a major effort to make it work in all cases in a backward compatible way.

Reply:
> it has to be %.0f kg with the blank

So '%.0f %%r.H.' would be syntactically correct (no space) but would nevertheless fail because there is no Units.RELATIVE_HUMIDITY defined.

Reply:
First Units.RELATIVE_HUMIDITY missing, second will probably fail because of the special treatment of %%. But that may be fixable.

Reply:
I provided a PR to at least show the channel type state description pattern when linking a new item to a channel. See https://github.com/openhab/openhab-webui/pull/2485.
It is imperfect because of the conceptual separation between channel type and item. An item can be linked to multiple channels.

Reply:
> Units.RELATIVE_HUMIDITY missing

Yes. I am proposing to create it.

> will probably fail because of the special treatment of %%

Probably not since %% is unescaped to % quite early in the deserialozation process.

Reply:
>  An item can be linked to multiple channels.

I understand your point. But please tell me why you think the state pattern element in thing xml exists if not for the purposes of (helping) creating items? It seems to me that 455 binding developers may have been wasting their time writing state pattern elements if the whole concept has no purpose..

Reply:
The discussion has moved to the PR in core https://github.com/openhab/openhab-webui/pull/2485 so I shall mark this here as 'resolved'.